site/en/tutorials/audio/simple_audio.ipynb

```diff
@@ -1,987 +1,1026 @@
+    "    subset='both')\n",
+    "\n",
+    "label_names = np.array(train_ds.class_names)\n",
+    "print()\n",
+    "print(\"label names:\", label_names)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "cestp83qFnU5"
+   },
+   "source": [
+    "The dataset now contains batches of audio clips and integer labels. The audio clips have a shape of `(batch, samples, channels)`. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "3yU6SQGIFb3H"
+   },
+   "outputs": [],
+   "source": [
+    "train_ds.element_spec"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "ppG9Dgq2Ex8R"
+   },
+   "source": [
+    "This dataset only contains single channel audio, so use the `tf.squeeze` function to drop the extra axis:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "Xl-tnniUIBlM"
+   },
+   "outputs": [],
+   "source": [
+    "def squeeze(audio, labels):\n",
+    "  audio = tf.squeeze(audio, axis=-1)\n",
+    "  return audio, labels\n",
+    "\n",
+    "train_ds = train_ds.map(squeeze, tf.data.AUTOTUNE)\n",
+    "val_ds = val_ds.map(squeeze, tf.data.AUTOTUNE)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "DtsCSWZN5ILv"
+   },
+   "source": [
+    "The `utils.audio_dataset_from_directory` function only returns up to two splits. It's a good idea to keep a test set separate from your validation set.\n",
+    "Ideally you'd keep it in a separate directory, but in this case you can use `Dataset.shard` to split the validation set into two halves. Note that iterating over **any** shard will load **all** the data, and only keep its fraction. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "u5UEGsqM5Gss"
+   },
+   "outputs": [],
+   "source": [
+    "test_ds = val_ds.shard(num_shards=2, index=0)\n",
+    "val_ds = val_ds.shard(num_shards=2, index=1)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "xIeoJcwJH5h9"
+   },
+   "outputs": [],
+   "source": [
+    "for example_audio, example_labels in train_ds.take(1):  \n",
+    "  print(example_audio.shape)\n",
+    "  print(example_labels.shape)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "voxGEwvuh2L7"
+   },
+   "source": [
+    "Let's plot a few audio waveforms:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "dYtGq2zYNHuT"
+   },
+   "outputs": [],
+   "source": [
+    "label_names[[1,1,3,0]]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "8yuX6Nqzf6wT"
+   },
+   "outputs": [],
+   "source": [
+    "plt.figure(figsize=(16, 10))\n",
+    "rows = 3\n",
+    "cols = 3\n",
+    "n = rows * cols\n",
+    "for i in range(n):\n",
+    "  plt.subplot(3, 3, i+1)\n",
```

Comment:
```suggestion
    "  plt.subplot(rows, cols, i+1)\n",
```

Reply:
sorry to miss this out. I shall update it